sql_runner/snowflake_target.go

```diff
@@ -188,3 +211,36 @@ func stringify(row [][]byte) []string {
 	}
 	return line
 }
+
+// Blocking function to poll for the true status of a query which didn't return a result.
+func pollForQueryStatus(sft SnowFlakeTarget, queryId string) error {
+	// Get the snoflake driver and open a connection
+	sfd := sft.Client.Driver()
+	conn, err := sfd.Open(sft.Dsn)
+	if err != nil {
+		return errors.Wrap(err, "Failed to open connection to poll for query result.")
+	}
+	// Poll Snowflake for actual query status
+	for {
+		qStatus, err := conn.(sf.SnowflakeConnection).GetQueryStatus(context.Background(), queryId)
+
+		switch {
+		case err != nil && strings.Contains(err.Error(), "[RUNNING]"):
```

Comment:
Good spot, I'm not sure, I'll look into that now.

Reply:
Yup looks like it'll be better for us to match against `279301`, since it returns that code in every instance of a still running query, but the server status might be something else. Nice catch.